util/ifdtool/ifdtool.c

```diff
@@ -1506,17 +1506,219 @@ static void set_pchstrap(struct fpsba *fpsba, const struct fdbar *fdb, const int
 	fpsba->pchstrp[strap] = value;
 }
 
+static int ifd2_platform_get_hap_location(struct fpsba *fpsba)
+{
+	switch (platform) {
+	case PLATFORM_SKLKBL:
+		/* No exceptions, always strap 0 */
+		return 0;
+	case PLATFORM_IFD2:
+		printf("To change the AltMeDisable/HAP bit please provide the platform as parameter\n");
+		return -1;
+	case PLATFORM_CNL:
+		/*
+		 * This gets little tricky here.
+		 * 1. CNL/CFL/CML-LP strap 106 (FPSBA + 0x70), but
+		 *    FSPBA + 0x70 bits 15:4 set to 0x1b
+		 *    FPSBA + 0x80 strap has configurable bits,
+		 *    FPSBA + 0x82/0x83 bytes are set to zeros,
+		 *    FPSBA + 0x00 is set to zeros,
+		 * 2. CML-V strap 0 (FPSBA + 0x00) with bits 15:2 set to 0x00, but
+		 *    FSPBA + 0x00 bits 15:2 must be set to 0,
+		 *    FPSBA + 0x00 bits 23:22 must be set to 0x2,
+		 *    FPSBA + 0x70 is set to all zeros,
+		 *    FPSBA + 0x80 byte has configurable bits,
+		 *    FPSBA + 0x81/0x82 bytes are set to zeros.
+		 * 3. CML-H/S strap 123 (FPSBA + 0x80), but
+		 *    FSPBA + 0x80 bits 15:4 set to 0x1b
+		 *    FPSBA + 0x00/0x70 is set to all zeros.
+		 *
+		 * So we do like this:
+		 *
+		 * 1. Prioritize CML-V detection as strap 0 cannot be all zeros.
+		 * 2. Then heck if FSPBA + 0x70 is zero for CML-H/S, CNL/CFL/CML-LP
+		 *    cannot have FPSBA + 0x70 all zeros.
+		 */
+		if (fpsba->pchstrp[0] != 0) {
+			/* CML-V, one more safetycheck ot exclude other SKUs */
+			if (fpsba->pchstrp[0x70 / 4] == 0) {
+				printf("Detected Comet Lake V platform flash descriptor\n");
+				return 0;
+			}
+		} else {
+			/* Do safety check for CML-H/S to be sure */
+			if (fpsba->pchstrp[0x70 / 4] == 0) {
+				printf("Detected Comet Lake H/S platform flash descriptor\n");
+				return 0x80 / 4;
+			}
+			/* Do safety check for CNL/CFL/CML-LP to be sure */
+			if (((fpsba->pchstrp[0x70 / 4] >> 4) & 0xfff) == 0x1b) {
+				printf("Detected Cannon Lake/Coffee Lake/Comet Lake LP "
+				       "platform flash descriptor\n");
+				return 0x70 / 4;
+			}
+		}
+		printf("Unusual Cannon Lake/Coffee Lake/Comet Lake flash descriptor detected\n"
+		       "Not modifying flash descriptor\n");
+		return -1;
+	case PLATFORM_TGL:
```

Comment:
Or revise the platform names, since right now one can refer to multiple different platform codenames like TGL / RKL here

Reply:
coreboot does not even have support for RKL so adding a dedicated platform enum for it doesn't seem beneficial. Moreover basically RKL is the same as TGL-H. AS long as the code does what it needs to do, it is fine for me (and probably for many others as well).